components/uniset/src/bin/gen_properties.rs

```diff
@@ -0,0 +1,458 @@
+use std::char;
+use std::collections::{HashMap, HashSet};
+use std::env;
+use std::fs::File;
+use std::io::{self, BufRead, Error};
+use std::iter::Iterator;
+use std::panic;
+use std::path::Path;
+use std::u32;
+
+use icu_unicodeset::{UnicodeSet, UnicodeSetBuilder};
+
+//
+// Can run with command in root of icu_unicodeset crate:
+//   cargo run --bin genprops <ppucd-file>
+//
+
+fn read_lines<P>(filename: P) -> io::Result<io::Lines<io::BufReader<File>>>
```

Comment:
Suggestion: add a crate error type for IO errors that carries the filename, like I did in CldrJsonDataProvider.

Reply:
Done. A crate error type was added, but carrying the filename doesn't apply in the PPUCD Provider, as you explained later.